### PR TITLE
Add GetCurrentUsers method

### PR DESCRIPTION
**Pull request by ProgramFOX**

### CODE REVIEW

File: ChatExchange.Net/Room.cs [hunk 1]
```diff
@@ -300,6 +301,36 @@ public HashSet<User> GetPingableUsers()
...
+        /// <summary>
+        /// Fetches a list of all users that are currently in the room.
+        /// </summary>
+        public HashSet<User> GetCurrentUsers()
+        {
+            string html = RequestManager.Get(cookieKey, string.Format("http://chat.{0}/rooms/{1}/", Host, ID));
+            CQ doc = CQ.CreateDocument(html);
+            IDomObject obj = doc.Select("script")[3];
+            string script = obj.InnerText;
+            Regex findUsers = new Regex(@"CHAT\.RoomUsers\.initPresent\(\[(?<users>.+)\]\);", RegexOptions.Compiled | RegexOptions.Singleline);
```
Comment (ArcticEcho) [inline ChatExchange.Net/Room.cs]: These regexes would need to be instantiated outside of the method for them to benefit from the `Compiled` flag.

File: ChatExchange.Net/Room.cs [hunk 2]
```diff
@@ -300,6 +301,36 @@ public HashSet<User> GetPingableUsers()
...
+            var lines = usersJs.Split(new char[] { '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                               .Select(x => x.Trim())
+                               .Where(x => !string.IsNullOrEmpty(x));
+            // The .Where call is to avoid lines who are whitespace-only, those do not get removed by RemoveEmptyEntries.
+
+            var users = new HashSet<User>();
+            Regex re = new Regex(@"id: (?<id>\d+)", RegexOptions.Compiled);
+            foreach (string line in lines)
+            {
+                Console.WriteLine(line);
```
Comment (ArcticEcho) [inline ChatExchange.Net/Room.cs]: And we probably don't need to print while we're parsing.

File: ChatExchange.Net/Room.cs [hunk 3]
```diff
@@ -300,6 +301,36 @@ public HashSet<User> GetPingableUsers()
...
+                string id = re.Match(line).Groups["id"].Value;
+                Console.WriteLine(id);
+                int userID = int.Parse(id);
```
Comment (ArcticEcho) [inline ChatExchange.Net/Room.cs]: Finally, being consistent with the use of `var` would be a bonus.

**Comment by ProgramFOX** [comment]
Done, done, and done.

**Comment by ArcticEcho** [comment]
Nice work, thanks!

**Pull request merged**